backend/Origam.Workbench.Services/DataLookupService.cs

```diff
@@ -605,31 +605,18 @@ public IMenuBindingResult GetMenuBinding(Guid lookupId, object value)
 
 		public NewRecordScreenBinding GetNewRecordScreenBinding(AbstractDataLookup lookup)
 		{
-			IParameterService param =
-				ServiceManager.Services.GetService(typeof(IParameterService)) as IParameterService;
+			IParameterService parameterService = ServiceManager.Services.GetService<IParameterService>();
 			IOrigamAuthorizationProvider authorizationProvider = SecurityManager.GetAuthorizationProvider();
 			IPrincipal principal = SecurityManager.CurrentPrincipal;
 
-
-			var newRecordScreenBinding = lookup.ChildItems
+			return lookup.ChildItems
 				.ToGeneric()
 				.OfType<NewRecordScreenBinding>()
-				.FirstOrDefault();
-
-			if (newRecordScreenBinding == null)
-			{
-				return null;
-			}
-
-			bool isAvailable = authorizationProvider.Authorize(principal, newRecordScreenBinding.AuthorizationContext)
-			                    && authorizationProvider.Authorize(principal, newRecordScreenBinding.MenuItem.AuthorizationContext)
-			                    && param.IsFeatureOn(newRecordScreenBinding.MenuItem.Features);
-			if (!isAvailable)
-			{
-				return null;
-			}
-
-			return newRecordScreenBinding;
+				.FirstOrDefault(binding => 
+					authorizationProvider.Authorize(principal, binding.AuthorizationContext)
+					&& authorizationProvider.Authorize(principal, binding.MenuItem.AuthorizationContext)
+					&& parameterService.IsFeatureOn(binding.MenuItem.Features)
```

Comment:
We should also check if the menu item is not ready only. Then we should not allow. It can be caused either by setting the menu item to `IsReadOnly = true` or by having a read only role on the menu (see how we generate the menu for the frontend).